packages/contracts/test/framework/plugin/plugin-setup-processor.ts

```diff
@@ -1769,236 +2309,93 @@ describe.skip('Plugin Setup Processor', function () {
-): Promise<{
-  updatedHelpers: string[];
-  permissions: PermissionOperation[];
-  initData: BytesLike;
-}> {
-  const pluginUpdateParams = {
-    plugin: proxy,
-    pluginSetupRepo: pluginRepo.address,
-    currentPluginSetup: currentVersionSetup.address,
-    newPluginSetup: newVersionSetup.address,
-  };
-
-  const prepareUpdateTx = await psp.prepareUpdate(
-    targetDao.address,
-    pluginUpdateParams,
+  data: BytesLike = EMPTY_DATA
+) {
+  const {initData} = await updatePlugin(
+    psp,
+    targetDao,
+    proxy,
+    currentVersionTag,
+    newVersionTag,
+    pluginRepo,
     currentHelpers,
-    EMPTY_DATA
+    data
   );
 
-  const prepareUpdateEvent = await findEvent(
-    prepareUpdateTx,
-    EVENTS.UpdatePrepared
-  );
-  expect(prepareUpdateEvent).to.not.equal(undefined);
-
-  const {
-    updatedHelpers: updatedHelpers,
-    permissions: permissions,
-    initData: initData,
-  } = prepareUpdateEvent.args;
+  const PluginRepoFactory = await ethers.getContractFactory('PluginRepo');
+  const repo = PluginRepoFactory.attach(pluginRepo);
 
-  // Grant the `UPGRADE_PLUGIN_PERMISSION_ID` permission to the plugin setup processor
-  await targetDao.grant(proxy, psp.address, UPGRADE_PLUGIN_PERMISSION_ID);
+  const currentVersion = await repo['getVersion((uint8,uint16))']({
+    release: currentVersionTag[0],
+    build: currentVersionTag[1],
+  });
+  const newVersion = await repo['getVersion((uint8,uint16))']({
+    release: newVersionTag[0],
+    build: newVersionTag[1],
+  });
 
-  const applyUpdateTx = await psp.applyUpdate(
-    targetDao.address,
-    proxy,
-    newVersionSetup.address,
-    pluginRepo.address,
-    initData,
-    permissions
+  const PluginSetupFactory = await ethers.getContractFactory(
+    'PluginUUPSUpgradeableSetupV1Mock'
   );
 
-  // Revoke the `UPGRADE_PLUGIN_PERMISSION_ID` permission to the plugin setup processor
-  await targetDao.revoke(proxy, psp.address, UPGRADE_PLUGIN_PERMISSION_ID);
-
-  const appliedUpdateEvent = await findEvent(
-    applyUpdateTx,
-    EVENTS.UpdateApplied
+  const currentPluginSetup = PluginSetupFactory.attach(
+    currentVersion.pluginSetup
   );
-  expect(appliedUpdateEvent).to.not.equal(undefined);
+  const newPluginSetup = PluginSetupFactory.attach(newVersion.pluginSetup);
 
   // If the base contracts don't change from current and new plugin setups,
   // PluginSetupProcessor shouldn't call `upgradeTo` or `upgradeToAndCall`
   // on the plugin. The below check for this still is not 100% ensuring,
   // As function `upgradeTo` might be called but event `Upgraded`
   // not thrown(OZ changed the logic or name) which will trick the test to pass..
-  const currentImpl = await currentVersionSetup.getImplementationAddress();
-  const newImpl = await newVersionSetup.getImplementationAddress();
+  const currentImpl = await currentPluginSetup.getImplementationAddress();
+  const newImpl = await newPluginSetup.getImplementationAddress();
+
+  // TODO: Giorgi Not working..
```

Comment:
Better remove the comments and push them to a separate branch and create a follow-up task.

Reply:
https://github.com/aragon/core/pull/242/commits/74729dae32addd7008613ddca878730aa50ed628